### PR TITLE
Add tests for freight-cache with apt_clean, fix bug with --keep

### PR DESCRIPTION
**Pull request by domcleal**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -82,7 +82,10 @@ gh-pages: man
 	git checkout -q master
 
 test/tmp/bats:
-	git clone https://github.com/sstephenson/bats test/tmp/bats
+	git clone --depth 1 https://github.com/sstephenson/bats test/tmp/bats
```
Comment (mmoll) [inline Makefile]: maybe add .git here?

File: Makefile [hunk 2]
```diff
@@ -82,7 +82,10 @@ gh-pages: man
...
+
+test/tmp/bats-assert:
+	git clone --depth 1 https://github.com/jasonkarns/bats-assert test/tmp/bats-assert
```
Comment (mmoll) [inline Makefile]: and there?

**Comment by mmoll** [comment]
What changed in `test/fixtures/test_1.0_all.deb`? :+1: otherwise, tests FTW!

File: Makefile (hunk 1 above)
Comment (domcleal) [inline Makefile]: Thanks, fixed.

**Comment by domcleal** [comment]
> What changed in test/fixtures/test_1.0_all.deb?

The old one contained another .deb and .rpm I'd erroneously included using fpm - now it's a properly empty deb!

**Comment by mmoll** [comment]
> The old one contained another .deb and .rpm I'd erroneously included using fpm

I see, :+1:

**Comment by mmoll** [comment]
merged, thanks @domcleal!

**Pull request merged**